1.raspberry-pi-stuff/octoberry/octoberry/__main__.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+import sys
+import re
+from pygame import mixer
+from urllib.parse import urlparse
+import os
+import requests
+
+BACKEND_URL = "http://localhost:5000" # TODO: change this to the backend url
+
+def play_sound(sound_url, cached_sound_path: None | str=None) -> None:
+    '''
+    Play sound from url, if cached_sound_path is provided it will be used to cache the sound
+
+    Args:
+        sound_url (str): url to sound file
+        cached_sound_path (None | str, optional): path to cache the sound file. Defaults to None.
+
+    Returns:
+        None
+    '''
+    if cached_sound_path is None:
+        parsed_url = urlparse(sound_url)
+        cached_sound_path = os.path.join(os.getcwd(), os.path.basename(parsed_url.path))
+
+    if not os.path.exists(cached_sound_path):
+        print("Downloading sound...")
+        r = requests.get(sound_url, allow_redirects=True)
+        open(cached_sound_path, 'wb').write(r.content)
+        print("Sound downloaded")
+
+    print("Playing sound...")
+    mixer = mixer.init()
+    mixer.music.load(cached_sound_path)
```

Comment:
I felt like using pygame here is a little bit weird and prefer something like python-vlc

apparently from all the options suggested in a relevant stack overflow this was according to the comments the most cross-platform and works way

Reply:
there should be native-ish ways of playing audio...? aplay or mpv..? but it all works.

as long as it supports mp3s and plays well with the raspi audio hardware output, we're good!!